src/components/MovieShowPage/MovieShowPage.js

```diff
@@ -1,21 +1,74 @@
-import React from 'react'
+import React, { Component } from 'react'
 
-const MovieShowPage = (props) => {
-  console.log(props.movie)
-  return (
-  <section className='movie-show-page'>
-    <img className='background' src={props.movie.backdrop_path} alt={props.movie.title + 'backdrop'}/>
-    <div className="movie-section">
-      <img className='main-poster' src={props.movie.poster_path}  alt={props.movie.title + 'poster'}/>
-      <div className='movie-info'>
-        <h1>{props.movie.title}</h1>
-        <h2>Release Date: {props.movie.release_date} </h2>
-        <h3>Rating: {props.movie.average_rating} </h3>
-        <p>{props.movie.overview}</p>
-      </div>
-    </div>
-  </section>
-)
+class MovieShowPage extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      rating: ''
+    }
+  }
+
+  submitRating = () => {
+    if (!Number.isInteger(+this.state.rating) || this.state.rating < 1 || this.state.rating > 10) {
+      alert('The number can only be a whole number between 1 and 10');
```

Comment:
Nice error handling! This is a great example of something you could write a sad path test for. Rather than making an alert pop up, I would suggest conditionally rendering something on the DOM. It's a better user experience and, doesn't steal focus, and is easier to test for.

Reply:
Good to know!